src/H5Zzfp.c

```diff
@@ -480,6 +497,39 @@ get_zfp_info_from_cd_values(size_t cd_nelmts, unsigned int const *cd_values,
     return retval;
 }
 
+static int
+zfp_codec_version_mismatch(
+    unsigned int zfpver_from_cd_val_data_in_file,
+    unsigned int zfpcodec_from_cd_val_data_in_file)
+{
+    int writer_codec;
+    int reader_codec;
+
+    /* We switched from 3 to 4 digit hex version numbers in H5Z-ZFP
+       1.1.0 when ZFP was at version 1.0.0. So, anything we read from
+       the file that has two leading zeros needs to be shifted up 4 bits
+       to compare correctly */
+    if (zfpver_from_cd_val_data_in_file < 0x00FF)
```

Comment:
Not that it matters in practice, but for correctness and readability, I think `0x00FF` ought to be `0x0100` the way the inequality is written.  Similarly with `ZFP_VERSION_NO < 0x04FF` below.